lib/monomorphic.mli

```diff
@@ -0,0 +1,12 @@
+val pp : Format.formatter -> t -> unit
+  (** Pretty printer, useful for debugging *)
+
+val show : t -> string
+  (** Convert value to string, useful for debugging *)
+
+val equal : t -> t -> bool
+  (** [equal a b] is the monomorphic equality.
+      Determines whether two JSON values are considered equal. In the case of
+      JSON objects, the order of the keys does not matter. In case of duplicate
+      keys, these have to exist in both objects to be considered equal.
```

Comment:
Yes, that's exactly the point I'm discussing.

Correctness in this context is an odd thing, because whether duplicate keys are valid or not does not seem to be clear. Some implementations pick the key parsed last, so in such case `{"a": 42, "a": 23}` would not be equal to `{"a": 23, "a": 42}` in any case because the parsed representation doesn't even include the duplicate. Yojson does allow that and also writes them out in the same order.

The more I think about it the more it seems to me like we should consider them equal if they are duplicated only if they are in the same order, which I guess is exactly what we get with `stable_sort`.